test/analytics/fixtures/plugins/analytics_ftr_helpers/public/plugin.ts

```diff
@@ -21,25 +22,10 @@ export class AnalyticsFTRHelpers implements Plugin {
       setOptIn(optIn: boolean) {
         analytics.optIn({ global: { enabled: optIn } });
       },
-      getEvents: async (takeNumberOfEvents, eventTypes = []) =>
-        firstValueFrom(
-          this.events$.pipe(
-            filter((event) => {
-              if (eventTypes.length > 0) {
-                return eventTypes.includes(event.event_type);
-              }
-              return true;
-            }),
-            take(takeNumberOfEvents),
-            toArray(),
-            // Sorting the events by timestamp... on CI it's happening an event may occur while the client is still forwarding the early ones...
-            map((_events) =>
-              _events.sort(
-                (a, b) => new Date(a.timestamp).getTime() - new Date(b.timestamp).getTime()
-              )
-            )
-          )
-        ),
+      getEvents: async (takeNumberOfEvents, options) =>
+        fetchEvents(this.events$, takeNumberOfEvents, options),
+      getEventCount: async (options) =>
+        (await fetchEvents(this.events$, Number.MAX_SAFE_INTEGER, options)).length,
```

Comment:
This is a bit odd, because if called as is, it will never return.
Maybe better to also set a default timeout of `1s`?